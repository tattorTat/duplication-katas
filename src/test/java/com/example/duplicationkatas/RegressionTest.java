package com.example.duplicationkatas;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

class RegressionTest {
    @Test
    void testMonkey() {
        LessonOne song = new LessonOne();
        song.singMonkeysJumpingOnTheBed();
        Approvals.verify(song.song);
    }

    @Test
    void testOldMan() {
        LessonTwo song = new LessonTwo();
        song.singThisOldMan();
        Approvals.verify(song.song);
    }

    @Test
    void testCheers() {
        LessonThree song = new LessonThree();
        song.singCheers();
        Approvals.verify(song.song);
    }

    @Test
    void testNames() {
        LessonFour song = new LessonFour();
        String[] names = {"Llewellyn", "Samatha", "Tomas", "Emilia"};
        song.singNames(1, names);
        song.singNames(2, names);
        song.singNames(3, names);
        Approvals.verify(song.song);
    }

    @Test
    void testWalk() {
        LessonFive song = new LessonFive();
        song.singWalkingSong();
        Approvals.verify(song.song);
    }

}