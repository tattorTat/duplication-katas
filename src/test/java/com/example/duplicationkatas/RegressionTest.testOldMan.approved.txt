This old man, he played one
He played knick-knack on my thumb
With a knick-knack paddywhack
Give the dog a bone
This old man came rolling home
This old man, he played two
He played knick-knack on my shoe
With a knick-knack paddywhack
Give the dog a bone
This old man came rolling home
